### PR TITLE
Run Adabot On Travis

### LINKED ISSUE #16: Run on Travis
Move off the computer at my house onto Travis.

https://docs.travis-ci.com/user/cron-jobs/

### PR DESCRIPTION
**Pull request by sommersoft**
Fixes #16.

## Adabot is ready to fly on her own! 🤖 🚀 

**This _will_ require some external setup.**

### Travis Settings
- Set the following environment variables:
    - ADABOT_GITHUB_ACCESS_TOKEN
    - ADABOT_TRAVIS_ACCESS_TOKEN
    - ARTIFACTS_KEY=(AWS access key id)
    - ARTIFACTS_SECRET=(AWS secret access key)
    - ARTIFACTS_BUCKET=(S3 bucket name)
    - ARTIFACTS_REGION (if it isn't the default "us-east-1")
- Turn off:
    - Build pushed branches
    - Build pushed pull requests
- Add a cron job to run `daily`

### AWS S3 Bucket
I've never worked with any AWS buckets, so these are guesses:

- Create `/adabot` directory, if needed. Travis will push the artifacts there (currently)
- Allow Adabot's Travis write access

## RFC 
@tannewt please give this a good look. I've used both the bundle's and core's `.travis.yml`s to work through this. However, it was also done with a lot of documentation reading.

I ran the `circuitpython_libraries` script on Travis with both pushed code and crons. Didn't run the `circuipython_bundle` for obvious reasons, and `circuitpython_libraries_download_stats` wasn't merged yet. Also, didn't push artifacts to AWS, but did have Travis deploy to releases successfully (since deleted). 

Travis builds can be viewed here: https://travis-ci.com/sommersoft/adabot/builds/90934238.

### CODE REVIEW

**Comment by tannewt** [comment]
I got credentials and am running a manual build now. https://travis-ci.org/adafruit/adabot/builds/455659198

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+env:
+    # LIB_CHECK_FILE is for circuitpython_libraries.py output
+    # LIB_DL_STATS_FILE is for future Bundle and PyPi download stats script
+  - LIB_CHECK_FILE='bin/adabot/library_report_'`date +%Y%m%d`'.txt' LIB_DL_STATS_FILE='bin/adabot/library_download_stats_'`date +%Y%m%d`'.txt'
+
+# Will need AWS S3 credentials added to the Travis settings
+addons:
+  artifacts:
+  paths:
```
Comment (tannewt) [inline .travis.yml]: This need to be indented?

**Comment by tannewt** [comment]
Looks like its uploading all of the git stuff it downloaded. I've stopped the build and will disable the cron for now. The upload worked. :-)

File: .travis.yml (hunk 1 above)
Comment (sommersoft) [reply .travis.yml]: yeah, i think so. oops.

**Pull request merged**